packages/runtime/inferno-hooks/__tests__/hooks.test.tsx

```diff
@@ -170,3 +167,73 @@ test('use memo is only called when its watch list changes', () => {
 
   expect(fx).toHaveBeenCalledTimes(2);
 });
+
+test('context binded to state', () => {
+  interface ConfigContextValue {
+    rtlEnabled?: boolean;
+  }
+  const ConfigContext = createContext<ConfigContextValue | undefined>(
+    undefined,
+  );
+  const contextChildren = () => {
+    const config = useContext(ConfigContext);
+    return (<span id="context">{config?.rtlEnabled && 'rtlEnabled'}</span>);
+  };
+  const contextProvider = () => {
+    const [rtlState, changeRtlState] = useState(false);
+    return (
+      <div>
+        <button type="button" id="b1" onClick={() => { changeRtlState(!rtlState); }}>11</button>
+        <ConfigContext.Provider value={{ rtlEnabled: rtlState }}>
+          <HookComponent renderFn={contextChildren} />
+        </ConfigContext.Provider>
+      </div>
+    );
+  };
+  const rendered = util.renderIntoContainer(<HookComponent renderFn={contextProvider} />);
+  const [button] = util.scryRenderedDOMElementsWithTag(rendered, 'button');
+  const [contextChildrenValue] = util.scryRenderedDOMElementsWithTag(rendered, 'span');
+
+  expect(contextChildrenValue.innerHTML).toBe('');
+  emit('onClick', button);
+  expect(contextChildrenValue.innerHTML).toBe('rtlEnabled');
+});
+
+test.only('useRef with method call (useImperativeHandle)', () => {
+  const mockFunc = jest.fn();
+
+  const childrenComponent = (ref: any) => (props: any) => {
```

Comment:
Because we need to save ref instance later 
`function ChildrenComponentHook(props: any, ref: any) {
    return (
      <HookComponent
        renderFn={
        childrenComponent(ref)
      }
        renderProps={props}
      />
    );
  }`
  If we just write (props, ref) we will need to add ref property to HookComponent just as renderProps